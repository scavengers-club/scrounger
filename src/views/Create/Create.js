import React, { useState } from 'react';
import { useHistory } from 'react-router';
import { createAdventure } from '../../services/adventures';

// import { uploadAdventureImage } from '../../services/adventures';

export default function Create({ currentUser }) {
  const [adventure, setAdventure] = useState({});
  const history = useHistory();
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await createAdventure(adventure);
      confirm('are you ready to submit this adventure?');
      history.push(`/user/${currentUser.id}`);
    } catch {
      alert('something went wrong!');
    }
  };
  //   const uploadFile = (e) => {
  //     uploadAdventureImage(currentUser.id, e.target.files[0]);
  //   };

  return <div></div>;
}
