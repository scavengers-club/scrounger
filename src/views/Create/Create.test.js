import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Create from './Create';

test('log in should render ', () => {
  const { container } = render(
    <MemoryRouter>
      <Create currentUser={{ id: 'ab0abbd1-29ad-414e-a8f2-343377f6b2bc' }} />{' '}
    </MemoryRouter>
  );
  screen.getAllByDisplayValue('');

  expect(container).toMatchSnapshot();
});
