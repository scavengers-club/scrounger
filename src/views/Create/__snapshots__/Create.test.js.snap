// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log in should render  1`] = `
<div>
  <div>
    <form
      class="adventure-form"
    >
      <label>
        name:
        <input
          placeholder="name"
          required=""
          type="text"
          value=""
        />
      </label>
      <label>
        image:
        <input
          type="file"
        />
      </label>
      <img />
      <label>
        description:
        <textarea
          placeholder="description"
          required=""
          type="text"
        />
      </label>
      <label>
        hint 1:
        <textarea
          placeholder="hint 1"
          required=""
          type="text"
        />
      </label>
      <label>
        hint 2:
        <textarea
          placeholder="hint 2"
          required=""
          type="text"
        />
      </label>
      <label>
        hint 3:
        <textarea
          placeholder="hint 3"
          required=""
          type="text"
        />
      </label>
      <label>
        solution:
        <input
          placeholder="solution"
          required=""
          type="text"
          value=""
        />
      </label>
      <button>
        Submit
      </button>
    </form>
  </div>
   
</div>
`;
